platform/mbed_application.c

```diff
@@ -107,7 +118,12 @@ __asm static void start_new_application(void *sp, void *pc)
 void start_new_application(void *sp, void *pc)
 {
     __asm volatile (
+#if defined(__CORTEX_M23)
```

Comment:
This two-instruction sequence is not required... "mov r2, #0" will put 0 (zero-extended) into r2 on the M23.  We can remove the entire #if block.  MOVW/MOVT is useful for loading a full 32-value into a register without a literal pool... not the case here.

Reply:
Sorry - missed the commit comments!  Clearly AC6 is not accepting the inline assembly properly (I ran a local test) so perhaps we could use "movw r2, #0" for both conditions and still remove the #if block.

Reply:
@TacoGrandeTX I replace with `movw`/`movt` and remove `#if` block in https://github.com/ARMmbed/mbed-os/pull/6949/commits/3c14cb633222cbc42fc9c1ea257da49db3a73d15.